src/main/java/org/pac4j/vertx/handler/impl/RequiresAuthenticationHandler.java

```diff
@@ -138,6 +145,33 @@ public void handle(RoutingContext routingContext) {
 
     }
 
+    @Override
+    protected void authorise(final User user, final RoutingContext context) {
+        if (! (user instanceof Pac4jUser)) {
+            throw new TechnicalException("Wrong user type in authorise");
+        }
+
+        Handler<AsyncResult<Boolean>> authHandler = res -> {
+            if (res.succeeded()) {
+                if (res.result()) {
+                    context.next();
+                } else {
+                    context.fail(403);
```

Comment:
In other implementations, this is replaced by a `forbidden` method which returns a 403 error by default, but allows some customizations...

-> https://github.com/pac4j/j2e-pac4j/blob/master/src/main/java/org/pac4j/j2e/filter/RequiresAuthenticationFilter.java#L138